veggieseasons/lib/screens/list.dart

```diff
@@ -15,65 +15,53 @@ import 'package:veggieseasons/widgets/veggie_card.dart';
 class ListScreen extends StatelessWidget {
   List<Widget> _generateVeggieRows(List<Veggie> veggies, Preferences prefs,
       {bool inSeason = true}) {
-    final cards = List<Widget>();
-
-    for (Veggie veggie in veggies) {
-      cards.add(Padding(
-        padding: EdgeInsets.only(left: 16.0, right: 16.0, bottom: 24.0),
+    return veggies.map((veggie) {
+      return Padding(
+        padding: EdgeInsets.only(left: 16, right: 16, bottom: 24),
         child: FutureBuilder<Set<VeggieCategory>>(
             future: prefs.preferredCategories,
             builder: (context, snapshot) {
               final data = snapshot.data ?? Set<VeggieCategory>();
               return VeggieCard(
                   veggie, inSeason, data.contains(veggie.category));
             }),
-      ));
-    }
-
-    return cards;
+      );
+    }).toList();
   }
 
   @override
   Widget build(BuildContext context) {
     return CupertinoTabView(
       builder: (context) {
         String dateString = DateFormat("MMMM y").format(DateTime.now());
+
         final appState =
             ScopedModel.of<AppState>(context, rebuildOnChange: true);
         final prefs =
             ScopedModel.of<Preferences>(context, rebuildOnChange: true);
 
-        final rows = <Widget>[];
-
-        rows.add(
-          Padding(
-            padding: const EdgeInsets.fromLTRB(16.0, 24.0, 16.0, 16.0),
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.start,
-              children: [
-                Text(dateString.toUpperCase(), style: Styles.minorText),
-                Text('In season today', style: Styles.headlineText),
-              ],
-            ),
-          ),
-        );
-
-        rows.addAll(_generateVeggieRows(appState.availableVeggies, prefs));
-
-        rows.add(
-          Padding(
-            padding: const EdgeInsets.fromLTRB(16.0, 24.0, 16.0, 16.0),
-            child: Text('Not in season', style: Styles.headlineText),
-          ),
-        );
-
-        rows.addAll(_generateVeggieRows(appState.unavailableVeggies, prefs,
-            inSeason: false));
-
         return DecoratedBox(
           decoration: BoxDecoration(color: Color(0xffffffff)),
           child: ListView(
-            children: rows,
+            children: [
+              Padding(
+                padding: const EdgeInsets.fromLTRB(16, 24, 16, 16),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(dateString.toUpperCase(), style: Styles.minorText),
+                    Text('In season today', style: Styles.headlineText),
+                  ],
+                ),
+              ),
+              ..._generateVeggieRows(appState.availableVeggies, prefs),
```

Comment:
You don't have to do it now but you'll want to change this to ListView.builder at some point

Reply:
No, you're right that I should switch it. I've made the change in a subsequent commit.

Is there an easier way to do a ListView.builder with contents this varied other than manually doing a bunch of index math?

Reply:
Nothing I can think of unfortunately. Maybe some clever tricks with generators?